addons/mass_mailing/models/mass_mailing.py

```diff
@@ -80,92 +63,78 @@ class MassMailingContact(osv.Model):
+    unsubscription_date = fields.Datetime(string='Unsubscription Date')
+    message_bounce = fields.Integer(string='Bounce', help='Counter of the number of bounced emails for this contact.')
+
+    @api.onchange('opt_out')
+    def on_change_opt_out(self):
+        self.unsubscription_date = self.opt_out and fields.Datetime.now()
+
+    @api.model
+    def create(self, vals):
         if 'opt_out' in vals:
-            vals['unsubscription_date'] = vals['opt_out'] and fields.datetime.now() or False
-        return super(MassMailingContact, self).create(cr, uid, vals, context=context)
+            vals['unsubscription_date'] = vals['opt_out'] and fields.Datetime.now() or False
+        return super(MassMailingContact, self).create(vals)
 
-    def write(self, cr, uid, ids, vals, context=None):
+    @api.multi
+    def write(self, vals):
         if 'opt_out' in vals:
-            vals['unsubscription_date'] = vals['opt_out'] and fields.datetime.now() or False
-        return super(MassMailingContact, self).write(cr, uid, ids, vals, context=context)
+            vals['unsubscription_date'] = vals['opt_out'] and fields.Datetime.now() or False
+        return super(MassMailingContact, self).write(vals)
 
-    def get_name_email(self, name, context):
-        name, email = self.pool['res.partner']._parse_partner_name(name, context=context)
+    def get_name_email(self, name):
+        name, email = self.env['res.partner']._parse_partner_name(name)
         if name and not email:
             email = name
         if email and not name:
             name = email
         return name, email
 
-    def name_create(self, cr, uid, name, context=None):
-        name, email = self.get_name_email(name, context=context)
-        rec_id = self.create(cr, uid, {'name': name, 'email': email}, context=context)
-        return self.name_get(cr, uid, [rec_id], context)[0]
+    def name_create(self, name):
+        name, email = self.get_name_email(name)
+        contact = self.create({'name': name, 'email': email})
+        return contact.display_name
 
-    def add_to_list(self, cr, uid, name, list_id, context=None):
-        name, email = self.get_name_email(name, context=context)
-        rec_id = self.create(cr, uid, {'name': name, 'email': email, 'list_id': list_id}, context=context)
-        return self.name_get(cr, uid, [rec_id], context)[0]
+    @api.model
+    def add_to_list(self, name, list_id):
+        name, email = self.get_name_email(name)
+        contact = self.create({'name': name, 'email': email, 'list_id': list_id})
+        return contact.display_name
 
-    def message_get_default_recipients(self, cr, uid, ids, context=None):
+    def message_get_default_recipients(self):
         res = {}
-        for record in self.browse(cr, uid, ids, context=context):
+        for record in self:
             res[record.id] = {'partner_ids': [], 'email_to': record.email, 'email_cc': False}
         return res
```

Comment:
can we return in one line this way : 

 return dict((record.id, {'partner_ids': [], 'email_to': record.email, 'email_cc': False}) for record in self)